### DISCUSSION TITLE
Generate deployments page with script from JSON file

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-docs#160] [Ideas]
[The deployments page](https://github.com/sablier-labs/docs/blob/0ff9d328a925e9ac2af63268a56b2791dcb0ef3e/docs/contracts/v2/02-deployments.md#L4) has become rather verbose and hence difficult to maintain.

What if we instead kept the deployment addresses in a neat JSON file, and generated the deployments page using a Shell script?

CC @andreivladbrg @smol-ninja.

**Comment by smol-ninja** [comment]
I don't think it's worth the effort.

1. We would still have to manually update the JSON file with the chain ID and the newly deployed contract addresses.
2. Adding a section for a new deployment to the deployment page seems less effort than adding it to the JSON and running the script.

**Reply by PaulRBerg** [reply]
Fair enough.

I feel like modifying a JSON is easier than a Markdown table, but I agree that the overhead of the script well compensates for that.

**Comment by andreivladbrg** [comment]
We can improve the DX by generating the deployment file in the deploy multi-chain script to match the docs format and then running the `cp` CLI command. See points 2 & 5 [here](https://github.com/sablier-labs/v2-core/issues/971). 

wdyt?

**Reply by smol-ninja** [reply]
We can copy the artefacts and deployments json to the deployment folder but updating docs would require more work. For example, consider the following questions:

1. What if we deploy to a new chain? In docs, we would then just append it following the alphabetical order of chain names. How would the cli work in this case?
2. What if we just deploy a newer version of the periphery in the next release as we have done in the past (I suppose)? We would want to replace only the existing periphery sections. How would cli work in this case?

I think deployments may be conditional and its not necessary that we always deploy them on all the chains at the same time. Take the example of Blast and Zksync. However, I think I may have misinterpreted your suggestion so feel free to clarify if you like.

**Reply by andreivladbrg** [reply]
I agree that `cp` may not be the best solution to automatically move them to deployments `docs` file.

Tho, we can still print the deployments address in the file of the format we would like:

```md
### Core

| Contract                | Address                                     | Deployment                             |
| :---------------------- | :------------------------------------------ | :------------------------------------- |
| SablierV2LockupDynamic  | [0x...](https://<explorer_url>/address/0x...) | [core-<version>](github_url_<version>) |
| SablierV2LockupLinear   | [0x...](https://<explorer_url>/address/0x...) | [core-<version>](github_url_<version>) |
| SablierV2LockupTranched | [0x...](https://<explorer_url>/address/0x...) | [core-<version>](github_url_<version>) |
| SablierV2NFTDescriptor  | [0x...](https://<explorer_url>/address/0x...) | [core-<version>](github_url_<version>) |

```

**Reply by smol-ninja** [reply]
Good idea of printing them. Would be easier to copy.

**Reply by PaulRBerg** [reply]
Thanks for your feedback @smol-ninja. I will close this discussion now.
